cmd/util/cmd/atree_inlined_status/cmd.go

```diff
@@ -0,0 +1,391 @@
+	atreeInlinedPayloadCount int,
+	atreeNonInlinedPayloadCount int,
+	err error,
+) {
+
+	if len(payloads)/numOfPayloadPerJob < nWorkers {
+		nWorkers = len(payloads) / numOfPayloadPerJob
+	}
+
+	log.Info().Msgf("checking atree payload inlined status...")
+
+	if nWorkers <= 1 {
+		// Skip goroutine to avoid overhead
+		for _, p := range payloads {
+			isAtreeSlab, isInlined, err := util.IsPayloadAtreeInlined(p)
+			if err != nil {
+				return 0, 0, err
+			}
+
+			if !isAtreeSlab {
+				continue
+			}
+
+			if isInlined {
+				atreeInlinedPayloadCount++
+			} else {
+				atreeNonInlinedPayloadCount++
+			}
+		}
+		return
+	}
+
+	type job struct {
+		payloads []*ledger.Payload
+	}
+
+	type result struct {
+		atreeInlinedPayloadCount    int
+		atreeNonInlinedPayloadCount int
+	}
+
+	numOfJobs := (len(payloads) + numOfPayloadPerJob - 1) / numOfPayloadPerJob
+
+	jobs := make(chan job, numOfJobs)
+
+	results := make(chan result, numOfJobs)
+
+	g, ctx := errgroup.WithContext(context.Background())
+
+	// Launch goroutine to check atree register inlined state
+	for i := 0; i < nWorkers; i++ {
+		g.Go(func() error {
+			for job := range jobs {
+				var result result
+
+				for _, p := range job.payloads {
+					isAtreeSlab, isInlined, err := util.IsPayloadAtreeInlined(p)
+					if err != nil {
+						return err
+					}
+
+					if !isAtreeSlab {
+						continue
+					}
+
+					if isInlined {
+						result.atreeInlinedPayloadCount++
+					} else {
+						result.atreeNonInlinedPayloadCount++
+					}
+				}
+
+				select {
+				case results <- result:
+				case <-ctx.Done():
+					return ctx.Err()
+				}
+			}
+			return nil
+		})
+	}
+
+	// Launch goroutine to wait for workers and close output channel
+	go func() {
+		_ = g.Wait()
+		close(results)
+	}()
+
+	// Send job to jobs channel
+	payloadStartIndex := 0
+	for {
+		if payloadStartIndex == len(payloads) {
+			close(jobs)
+			break
+		}
+
+		endIndex := payloadStartIndex + numOfPayloadPerJob
+		if endIndex > len(payloads) {
+			endIndex = len(payloads)
+		}
+
+		jobs <- job{payloads: payloads[payloadStartIndex:endIndex]}
+
+		payloadStartIndex = endIndex
+	}
+
+	// Gather results
+	for result := range results {
+		atreeInlinedPayloadCount += result.atreeInlinedPayloadCount
+		atreeNonInlinedPayloadCount += result.atreeNonInlinedPayloadCount
+	}
+
+	log.Info().Msgf("waiting for goroutines...")
+
+	if err := g.Wait(); err != nil {
+		return 0, 0, err
+	}
+
+	return atreeInlinedPayloadCount, atreeNonInlinedPayloadCount, nil
+}
+
+type stateStatus struct {
+	InputPayloadFile            string `json:",omitempty"`
+	InputState                  string `json:",omitempty"`
+	InputStateCommitment        string `json:",omitempty"`
+	TotalPayloadCount           int
+	SamplePayloadCount          int
+	AtreeInlinedPayloadCount    int
+	AtreeNonInlinedPayloadCount int
+}
+
+func readTrie(dir string, targetHash flow.StateCommitment) ([]*ledger.Payload, error) {
```

Comment:
`readTrie` and `parseStateCommitment` are the same as in `diff-states`. Maybe move to `cmd/util/ledger/util` (or a more appropriate "common code" place) and re-use the functions in both this new command and the `diff-states` command